### PR TITLE
Merge vote packet with pbft block packet

### PR DESCRIPTION
**Pull request by JakubFornadel**
## Purpose

Work in progress:

- Remove pbft block packet
- send pbft blocks as optional value with each new vote

### CODE REVIEW

**Review by mfrankovi — APPROVED** [review]

File: libraries/core_libs/consensus/include/pbft/proposed_blocks.hpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   * @brief Cleanup all proposed PBFT blocks for the finalized round - 1. We must keep previous round proposed blocks
+   *        for voting purposes
+   * @param period
+   * @param round
+   */
+  void cleanupProposedPbftBlocksByRound(uint64_t period, uint64_t round);
+
+ private:
+  // <PBFT period, <PBFT round, <block hash, block>>>
+  std::map<uint64_t, std::map<uint64_t, std::unordered_map<blk_hash_t, std::shared_ptr<PbftBlock>>>> proposed_blocks_;
```
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/consensus/include/pbft/proposed_blocks.hpp:70]: I am deleting from from proposed_blocks_ based on old period and round so thats why map

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp [hunk 2]
```diff
@@ -157,6 +158,7 @@ bool VoteManager::addVerifiedVote(std::shared_ptr<Vote> const& vote) {
   }
 
   found_voted_value_it->second.first += weight;
+  db_->saveVerifiedVote(vote);
```
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:161]: uff, you are probably right. So is it the case that we want to save only votes that we generated ?

File: libraries/core_libs/consensus/include/pbft/pbft_manager.hpp [hunk 3]
```diff
@@ -601,11 +624,13 @@ class PbftManager : public std::enable_shared_from_this<PbftManager> {
   size_t step_ = 1;
   size_t startingStepInRound_ = 1;
 
-  std::optional<std::pair<blk_hash_t, uint64_t /* period */>> soft_voted_block_for_round_{};
+  // Block that node saw 2t+1 soft votes for
+  std::optional<std::pair<blk_hash_t, std::shared_ptr<PbftBlock>>> soft_voted_block_for_round_{};
 
-  std::optional<std::pair<blk_hash_t, uint64_t /* period */>> cert_voted_block_for_round_{};
+  // Block that node cert voted
+  std::optional<std::shared_ptr<PbftBlock>> cert_voted_block_for_round_{};
```
Comment (kstdl) [inline libraries/core_libs/consensus/include/pbft/pbft_manager.hpp:631]: Isn't this optional excess here?

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:161]: no this was just my mistake - this PR does not require to save them all :)

**Review by kstdl — APPROVED** [review]

File: tests/network_test.cpp [hunk 4]
```diff
@@ -155,22 +154,23 @@ TEST_F(NetworkTest, transfer_lot_of_blocks) {
...
-  uint64_t chain_size = 111;
-
-  nw2->getSpecificHandler<network::tarcap::PbftBlockPacketHandler>()->sendPbftBlock(nw1->getNodeId(), pbft_block,
-                                                                                    chain_size);
-
-  auto node2_id = nw2->getNodeId();
-  EXPECT_HAPPENS({10s, 200ms},
-                 [&](auto& ctx) { WAIT_EXPECT_EQ(ctx, nw1->getPeer(node2_id)->pbft_chain_size_, chain_size) });
-}
+// TODO: revisit
```
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply tests/network_test.cpp]: https://github.com/Taraxa-project/taraxa-node/pull/2012/commits/2bfd0e4606cc7d8d19de3790047ee5a464530636

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp]: https://github.com/Taraxa-project/taraxa-node/pull/2012/commits/22ae85387590d337eb306fb5407d86d225202387

File: libraries/core_libs/consensus/include/pbft/pbft_manager.hpp (hunk 3 above)
Comment (JakubFornadel) [reply libraries/core_libs/consensus/include/pbft/pbft_manager.hpp:631]: you are probably right. I will create another PR https://github.com/Taraxa-project/taraxa-node/issues/2034

**Pull request merged**